### DISCUSSION TITLE
Request Forwarding Timing

### DISCUSSION THREAD

**Discussion by EinavDanielDX** [kedacore/http-add-on#1114] [Q&A]
Hi!

I would like to know how often the interceptor checks the target service/deployment before forwarding requests from its queue.

For example:
I send an HTTP request to Service A. Deployment A is being scaled from 0 to 1, and the pod is ready within 10 seconds, yet the request is being held in the queue for over 30 seconds. Why is that?

**Comment by EinavDanielDX** [comment]
Answering here in case anybody else face this issue:
eventually what solved my problem was setting these env variable:
```
KEDA_RESPONSE_HEADER_TIMEOUT="120s" 
KEDA_HTTP_CONNECT_TIMEOUT="5s"   # Was 30s before, reducing this fixed the delay 
```

**Reply by linkvt** [reply]
Probably not relevant anymore, but we just released a fix for this in 0.12.0, there was an issue with the cold start requests and high connection timeouts.
